aspnetcore/mvc/models/model-binding.md

```diff
@@ -625,6 +625,9 @@ For more information, see [TryUpdateModelAsync](xref:data/ef-rp/crud#TryUpdateMo
 
 This attribute's name follows the pattern of model binding attributes that specify a data source. But it's not about binding data from a value provider. It gets an instance of a type from the [dependency injection](xref:fundamentals/dependency-injection) container. Its purpose is to provide an alternative to constructor injection for when you need a service only if a particular method is called.
 
+> [!TIP]
+> The binding will fail when obtaining an instance of a type, **not registered**, from the dependency injection container. If the service is expected to be **optional** then the parameter must be marked as nullable, including `?`, or set a default value. In this case, the method is responsible for the required check before the usage.
```

Comment:
```suggestion
> [!TIP]
> Binding fails when attempting to bind to a type that isn't registered with the dependency injection container or isn't a .NET type. If the service is **optional**,  the parameter must be one of the following:
>* Marked as [nullable](/dotnet/csharp/language-reference/builtin-types/nullable-value-types).
>* Have a default value.
>
> When using optional parameters, the app must check if the parameter is available.
```

This might be closer. @serpent5  can you help out.

Reply:
I've made an alternative suggestion below. I don't think this needs to be a tip, and I don't think we should link to the nullable docs as they're specific to value-types.